packages/evm/contracts/adapters/Proofcast/ProofcastAdapter.sol

```diff
@@ -0,0 +1,130 @@
+    error UnsupportedChainId(uint256 chainId);
+    error UnexpectedEventTopic(bytes32 topic);
+    error InvalidSender();
+    error InvalidMessageId(uint256 actual, uint256 expected);
+    error InvalidDestinationChainId(uint256 chainId);
+
+    event TeeSignerChanged(address newAddress);
+    event TeeSignerPendingChange(address newAddress, bytes attestation, uint256 gracePeriod);
+    event YahoInitialized(uint256 chainId, address yaho);
+
+    function initYaho(uint256 chainId, address yaho_) public onlyOwner {
+        if (chainId == block.chainid) {
+            revert NotUsableYaho();
+        }
+        yahos[chainId] = yaho_;
+        emit YahoInitialized(chainId, yaho_);
+    }
+
+    function setTeeSigner(bytes calldata pubKey, bytes memory attestation) public onlyOwner {
+        if (teeAddress == address(0)) {
+            // Setting the teeAddress the first time
+            teeAddress = _getAddressFromPublicKey(pubKey);
+            emit TeeSignerPendingChange(teeAddress, attestation, block.timestamp);
+            emit TeeSignerChanged(teeAddress);
+        } else {
+            // The new address will be set after a grace period of 48 hours
+            teeAddressNew = _getAddressFromPublicKey(pubKey);
+            teeAddressChangeGraceThreshold = block.timestamp + TEE_ADDRESS_CHANGE_GRACE_PERIOD;
+            emit TeeSignerPendingChange(teeAddressNew, attestation, teeAddressChangeGraceThreshold);
+        }
+    }
+
+    function verifyEventAndStoreHash(bytes calldata statement, bytes memory signature) public {
+        if (teeAddressNew != address(0) && block.timestamp > teeAddressChangeGraceThreshold) {
+            teeAddress = teeAddressNew;
+            teeAddressNew = address(0);
+            emit TeeSignerChanged(teeAddress);
+        }
+        if (teeAddress == address(0)) revert InvalidTeeSigner();
+        if (ECDSA.recover(sha256(statement), signature) != teeAddress) revert InvalidSignature();
+
+        // Supports only EVM events
+        bytes1 protocolId = statement[1];
+        if (protocolId != 0x00) revert UnsupportedProtocolId(protocolId);
+
+        (uint256 domain, uint256[] memory ids, bytes32[] memory hashes) = _checkEventAndDecodeData(statement);
+        _storeHashes(domain, ids, hashes);
+    }
+
+    function _checkEventAndDecodeData(
+        bytes calldata statement
+    ) internal view returns (uint256 domain, uint256[] memory ids, bytes32[] memory hashes) {
+        //  Statement format:
+        //    | version   | protocol   |  protocol_chain_id |   event id    | event_bytes |
+        //    | (1 byte)  | (1 byte)   |    (32 bytes)      |  (32 bytes)   |  varlen     |
+
+        uint16 offset = 2; // skip version, protocolId
+        domain = uint256(bytes32(statement[offset:(offset += 32)]));
+
+        if (yahos[domain] == address(0)) revert UnsupportedChainId(domain);
+
+        offset += 32; // skip the event id (32 bytes)
+        bytes memory eventBytes = statement[offset:];
```

Comment:
From the statement format, you should have the event id at this offset.

Reply:
isn't it skipped on line `95`?

Reply:
My bad, I overlooked `offset += 32` at line 91.
Resolving this and the other comment, which was also tied to the fact that I didn't spot the event id skipping.